### ISSUE TITLE
datadog-ipc-helper uses obsolete binary /memfd:spawn_worker_trampoline

### ISSUE THREAD

**Issue by moerkey** [liske/needrestart#287]
needrestart is complaining about the process datadog-ipc-helper from https://github.com/DataDog/dd-trace-php/releases/tag/0.90.0.

```
[main] eval /etc/needrestart/needrestart.conf
[main] needrestart v3.5
[main] running in root mode
[main] systemd detected
[main] vm detected
NEEDRESTART-VER: 3.5
[Core] #615 is a NeedRestart::Interp::Python
[Python] #615: source=/usr/bin/networkd-dispatcher
[main] #137881 uses obsolete binary /memfd:spawn_worker_trampoline
[main] #137881 is not a child
[main] #137881 exe => /memfd:spawn_worker_trampoline
[main] trying systemctl status
[main] #137881 is php8.0-fpm.service
[main] inside container or vm, skipping microcode checks
[Kernel] Linux: kernel release 5.15.0-88-generic, kernel version #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023
[Kernel/Linux] /boot/vmlinuz-5.15.0-88-generic => 5.15.0-88-generic (buildd@lcy02-amd64-058) #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 [5.15.0-88-generic]*
[Kernel/Linux] /boot/vmlinuz-5.15.0-87-generic => 5.15.0-87-generic (buildd@lcy02-amd64-011) #97-Ubuntu SMP Mon Oct 2 21:09:21 UTC 2023 [5.15.0-87-generic]
[Kernel/Linux] Expected linux version: 5.15.0-88-generic
NEEDRESTART-KCUR: 5.15.0-88-generic
NEEDRESTART-KEXP: 5.15.0-88-generic
NEEDRESTART-KSTA: 1
NEEDRESTART-SVC: php8.0-fpm.service
```

Using.
```
root@SERVER:~# cat /etc/issue
Ubuntu 22.04.3 LTS \n \l

root@SERVER:~# dpkg -l | grep fpm
ii  php7.3-fpm                            7.3.33-14+ubuntu22.04.1+deb.sury.org+1       amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php7.4-fpm                            1:7.4.33-8+ubuntu22.04.1+deb.sury.org+1      amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php8.0-fpm                            1:8.0.30-2+ubuntu22.04.1+deb.sury.org+1      amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)

root@SERVER:~# ls -lh /opt/datadog/dd-library/
total 20K
drwxr-xr-x 5 root root 4,0K Jul  6 06:19 0.89.0
drwxr-xr-x 5 root root 4,0K Jul 19 06:07 0.90.0
drwxr-xr-x 5 root root 4,0K Sep 25 12:07 0.91.2
drwxr-xr-x 5 root root 4,0K Nov  2 15:35 0.93.1
drwxr-xr-x 5 root root 4,0K Nov 14 09:06 0.93.2

root@SERVER:~# needrestart --version

needrestart 3.5 - Restart daemons after library updates.
...
# Yes, there is needrestart 3.6 already but the error is still the same.
```

I have checked https://github.com/liske/needrestart/issues/130 without luck but created a workaround inspired by https://github.com/liske/needrestart/issues/283.

Here is also a detailed post https://github.com/DataDog/dd-trace-php/issues/2357

**Comment by liske** [comment]
All `/memfd:` mappings are already ignored in git head: https://github.com/liske/needrestart/commit/6c87772bdc38091e9efbf4c59217fa502365dcf2

So this issue should not happen with needrestart >3.6

**Comment by liske** [comment]
Sorry this was a new issue which was not fixed in needrestart, yet. The fix will be part of needrestart 3.9.
